Bottom: b789541152cc00f83febf350e5c8844d43d04a12
Top:    90390f70eee3c67c604ecdd8a156725f59899704
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 17:47:37 +0000

polygon1.c: Fix possible case where we could return an invalid polygon upon allocation failure

This was a bug present in the poly_Boolean() routine. We need to set the output
polygon to NULL if we are going to return early on memory failure. Comment this,
and add a similar note to poly_Boolean_free(), where we already set the return
value (and hence there was no bug).


---

diff --git a/src/polygon1.c b/src/polygon1.c
index f476cc4..0403874 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2400,6 +2400,8 @@ poly_Boolean (const POLYAREA * a_org, const POLYAREA * b_org,
 {
   POLYAREA *a = NULL, *b = NULL;
 
+  *res = NULL;
+
   if (!poly_M_Copy0 (&a, a_org) || !poly_M_Copy0 (&b, b_org))
     return err_no_memory;
